### PR TITLE
bpo-42423: Accept single base class in PyType_FromModuleAndSpec()

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue42423

### CODE REVIEW

File: Modules/_hashopenssl.c [hunk 1]
```diff
@@ -2038,21 +2038,14 @@ hashlib_init_evpxoftype(PyObject *module)
...
     }
 
-    bases = PyTuple_Pack(1, state->EVPtype);
-    if (bases == NULL) {
-        return -1;
-    }
-
     state->EVPXOFtype = (PyTypeObject *)PyType_FromSpecWithBases(
-        &EVPXOFtype_spec, bases
+        &EVPXOFtype_spec, state->EVPtype
```
Comment (tiran) [inline Modules/_hashopenssl.c]:
```suggestion
        &EVPXOFtype_spec, (PyObject *)state->EVPtype
```

File: Objects/structseq.c [hunk 2]
```diff
@@ -526,13 +526,7 @@ PyStructSequence_NewType(PyStructSequence_Desc *desc)
...
     spec.slots = slots;
 
-    bases = PyTuple_Pack(1, &PyTuple_Type);
-    if (bases == NULL) {
-        PyMem_FREE(members);
-        return NULL;
-    }
-    type = (PyTypeObject *)PyType_FromSpecWithBases(&spec, bases);
-    Py_DECREF(bases);
+    type = (PyTypeObject *)PyType_FromSpecWithBases(&spec, &PyTuple_Type);
```
Comment (tiran) [inline Objects/structseq.c]:
```suggestion
    type = (PyTypeObject *)PyType_FromSpecWithBases(&spec, (PyObject *)(&PyTuple_Type));
```

**Review by tiran — CHANGES_REQUESTED** [review]

File: Doc/whatsnew/3.10.rst [hunk 3]
```diff
@@ -512,8 +512,12 @@ New Features
...
   (Contributed by Victor Stinner in :issue:`42262`.)
 
-* The :c:func:`PyType_FromModuleAndSpec` function now accepts NULL ``tp_doc``
-  slot.
+* The :c:func:`PyType_FromSpecWithBases` and :c:func:`PyType_FromModuleAndSpec`
+  functions now accept a single class as the *bases* argument.
+  (Contributed by Serhiy Storchaka in :issue:`42423`.)
+
+* The :c:func:`PyType_FromModuleAndSpec` and :c:func:`PyType_FromModuleAndSpec`
+  functions now accept ``NULL`` as the ``tp_doc`` slot.
```
Comment (encukou) [inline Doc/whatsnew/3.10.rst]: Please don't change the other entry.

**Comment by serhiy-storchaka** [comment]
I have made the requested changes; please review again.

**Review by tiran — APPROVED** [review]

**Comment by encukou** [comment]
Thank you!

**Pull request merged**